src/crystalla/matrix.cr

```diff
@@ -104,6 +121,14 @@ module Crystalla
       Matrix.new(added, number_of_rows, number_of_cols)
     end
 
+    def -
+      m = self.clone
+      m.values.size.times do |i|
```

Comment:
In fact `Matrix.new values.map(&.-), number_of_rows, number_of_cols` should be faster. Something similar could be done for the `+` method.